src/classes/AuthenticationSession.ts

```diff
@@ -45,41 +108,59 @@ export class AuthenticationSession {
   }
 
   private processPasswordResponse([password]: string[]): void {
+    if (!AuthenticationSession.sftpFusionAuthAppId) {
+      logger.error('SFTP application id missing. No refresh token would be returned');
+    }
     this.fusionAuthClient.login({
+      applicationId: AuthenticationSession.sftpFusionAuthAppId,
       loginId: this.authContext.username,
       password,
     }).then((clientResponse) => {
       switch (clientResponse.statusCode) {
-        case FusionAuthStatusCode.Success:
-        case FusionAuthStatusCode.SuccessButUnregisteredInApp:
+        case FusionAuthStatusCode.Success: {
           if (clientResponse.response.token !== undefined) {
             logger.verbose('Successful password authentication attempt.', {
               username: this.authContext.username,
             });
             this.authToken = clientResponse.response.token;
+            this.authTokenExpiresAt = clientResponse.response.tokenExpirationInstant ?? 0;
+            this.refreshToken = clientResponse.response.refreshToken ?? '';
             this.authContext.accept();
-            return;
+          } else {
+            this.authContext.reject();
           }
-          this.authContext.reject();
           return;
-        case FusionAuthStatusCode.SuccessNeedsTwoFactorAuth:
+        }
+        case FusionAuthStatusCode.SuccessButUnregisteredInApp: {
+          const userId: string = clientResponse.response.user?.id ?? '';
+          this.registerUserInApp(userId)
+            .then(() => { this.processPasswordResponse([password]); })
+            .catch((error) => {
+              logger.warn('Error during registration and authentication:', error);
+              this.authContext.reject();
```

Comment:
I forget if `reject` takes an optional message but if so we should communicate the reason for rejection (this should be done in the other locations too FWIW, I missed these when doing #212).

Reply:
I can add this in a separate commit once it's clear to me that it takes on optional message. 

For now, the type hints tell me it takes arrays of "authentication types".

https://github.com/PermanentOrg/sftp-service/blob/main/src/classes/SshConnectionHandler.ts#L37

Reply:
It looks like it does *NOT* take an additional message!  I'm marking this as resolved.